packages/generators/add-generator.ts

```diff
@@ -421,49 +448,57 @@ module.exports = class AddGenerator extends Generator {
 							return;
 						}
 						// Either we are adding directly at the property, else we're in a prop.theOne scenario
-						const actionMessage =
+						const actionMessage: string =
 							isDeepProp[1] === "other"
 								? `What do you want the key on ${action} to be? (press enter if you want it directly as a value on the property)`
 								: `What do you want the value of ${isDeepProp[1]} to be?`;
 
-						this.prompt([Input("deepProp", actionMessage)]).then(
-							deepPropAns => {
+						this.prompt([
+							Input("deepProp", actionMessage),
+						]).then(
+							(deepPropAns: {
+								deepProp: string,
+							}) => {
 								// The other option needs to be validated of either being empty or not
 								if (isDeepProp[1] === "other") {
-									let othersDeepPropKey = deepPropAns.deepProp
+									const othersDeepPropKey: string = deepPropAns.deepProp
 										? `What do you want the value of ${
 											deepPropAns.deepProp
 										  } to be?` // eslint-disable-line
 										: `What do you want to be the value of ${action} to be?`;
 									// Push the answer to the array we have created, so we can use it later
 									isDeepProp.push(deepPropAns.deepProp);
-									this.prompt([Input("deepProp", othersDeepPropKey)]).then(
-										deepPropAns => {
+									this.prompt([
+										Input("innerProp", othersDeepPropKey),
+									]).then(
+										(innerPropAns: {
+											innerProp,
+										}) => {
 											// Check length, if it has none, add the prop directly on the given action
 											if (isDeepProp[2].length === 0) {
 												this.configuration.config.item = action;
 												this.configuration.config.webpackOptions[action] =
-													deepPropAns.deepProp;
+													innerPropAns.innerProp;
 											} else {
 												// If not, we're adding to something like devServer.myProp
 												this.configuration.config.item =
 													action + "." + isDeepProp[2];
 												this.configuration.config.webpackOptions[action] = {
-													[isDeepProp[2]]: deepPropAns.deepProp
+													[isDeepProp[2]]: innerPropAns.innerProp,
 												};
 											}
 											done();
-										}
+										},
 									);
 								} else {
 									// We got the schema prop, we've correctly prompted it, and can add it directly
 									this.configuration.config.item = action + "." + isDeepProp[1];
 									this.configuration.config.webpackOptions[action] = {
-										[isDeepProp[1]]: deepPropAns.deepProp
+										[isDeepProp[1]]: deepPropAns.deepProp,
```

Comment:
Not sure how dynamic types are in typescript, but I think it's possible to reassign types or to make it `oneOf`

Reply:
oneOf as in it could be one of many types, but not `any`.